### ISSUE TITLE
Reusable slides - subfolders in data doesn't work

### ISSUE THREAD

**Issue by valvin1** [dzello/reveal-hugo#44]
First of all thank you for reveal-hugo :) 

i've started working with reveal-hugo very recently and i hope i've not missed something on this issue.

we planned to use reveal-hugo at work for our different support. For this I would like to have in a seperate repository reusables slides that I'll plug using git submodules in data folder.

it looks like today we can only import reusable slides in a file at the root folder of `data`. For example `./data/home.toml`with a variable `reusable` which leads to a slide content declared as `data.home.reusable` and works well.

In order to use a separate repo for global reusable slides and allowes user to create their owns i've tested to import my repo in a subfolder `./data/global/home.toml` which leads to a content declared as `data.global.home.reusable`. 

This subfolder looks compatible with hugo standard but unfortunately doesn't work.

Is there a tip to make it works ?

For my usecase I've tried to trick using symbolic links but unfortunately it is not supported in data folder by hugo :(

**Comment by dzello** [comment]
Cool use case for the reusable slides feature!

The code that constructs the path to look in the data files is here: https://github.com/dzello/reveal-hugo/blob/master/layouts/shortcodes/slide.html#L6

I can see what the issue might be - only index 0 and 1 are used. Try replacing it with:

```
  {{- $html := index .Page.Site.Data (index $lookup 0) (index $lookup 1) (index $lookup 2) | markdownify -}}
```

And see if that works for you. If so, I can think about how we could get that into the code permanently.

**Comment by valvin1** [comment]
Thank you @dzello 

I've just tested it and it works. Using this method now all reusable slides should be in `data/<subfolder>/<file>.toml` and it corresponds to my usecase :)

If you are interested I can have a look to add a settings to switch from reusable slides at root data root folder to subfolder ?

**Comment by dzello** [comment]
Good, glad that worked!

I feel like we can just change that line of code to be smarter about how much nesting is in the path, so it doesn't have to be configured. Would you be interested in giving that a try?

**Comment by valvin1** [comment]
Yes of course :)

It looks this piece of code is working but I'm not sure it is the good way to do it 

```
{{- if $content -}}
  {{- $lookup := split $content "." }}
  {{ $markdownContent := .Page.Site.Data }}
  {{- range $dataPath := $lookup -}}
    {{- $markdownContent = index $markdownContent $dataPath -}}
  {{- end -}}
  {{- $html := $markdownContent | markdownify -}}
  {{- $slides := split $html "<hr />" -}}
  {{- range $slides -}}
    {{- $scratch.Add "slides" . -}}
  {{- end -}}
{{- else -}}
```

**Comment by dzello** [comment]
Yep, I think that's right. Merged in #45, thanks for the contribution!
